plugins/all_test.go

```diff
@@ -0,0 +1,114 @@
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// exceptionMap holds those plugins which differ in conventions when defining plugins.
+// Most plugins follow the convention of pkg-name to plugin-name.
+// For ex, Pivot processor pkg github.com/influxdata/telegraf/plugins/processors/pivot maps directly to
+// the last element of the pkg i.e "pivot"
+// But in case of "aws_ec2" processor, the pkg is defined as "github.com/influxdata/telegraf/plugins/processors/aws/ec2".
+// This ensure package names are not tied with plugin names.
+// it should be of the form <pkg-name>: <plugin-name>
+var exceptionMap = map[string]string{
+	"github.com/influxdata/telegraf/plugins/processors/aws/ec2": "aws_ec2",
+}
+
+func Test_AllPlugins(t *testing.T) {
+	pluginDirs := []string{"aggregators", "inputs", "outputs", "parsers", "processors", "secretstores"}
+	for _, dir := range pluginDirs {
+		testPluginDirectory(t, dir)
+	}
+}
+
+func testPluginDirectory(t *testing.T, directory string) {
+	allDir := filepath.Join(directory, "all")
+	err := filepath.WalkDir(allDir, func(goPluginFile string, d fs.DirEntry, walkErr error) error {
+		require.NoError(t, walkErr)
+		if d.IsDir() || strings.HasSuffix(d.Name(), "_test.go") {
+			return nil
+		}
+		parseSourceFile(t, goPluginFile, directory)
+		return nil
+	})
+	require.NoError(t, err)
+}
+
+func parseSourceFile(t *testing.T, goPluginFile string, pluginCategory string) {
+	fset := token.NewFileSet()
+	node, err := parser.ParseFile(fset, goPluginFile, nil, parser.ParseComments)
+	require.NoError(t, err)
+
+	for _, cg := range node.Comments {
+		for _, comm := range cg.List {
+			if !strings.HasPrefix(comm.Text, "//go:build") {
+				continue
+			}
+			file := filepath.Base(goPluginFile)
+			testName := fmt.Sprintf("%v-%v", pluginCategory, strings.TrimSuffix(file, ".go"))
+			t.Run(testName, func(t *testing.T) {
+				plugin := resolvePluginFromImports(t, node.Imports)
+				testBuildTags(t, comm.Text, pluginCategory, plugin)
+			})
+		}
+	}
+}
+
+func resolvePluginFromImports(t *testing.T, imports []*ast.ImportSpec) string {
+	// should contain one or more import statements
+	require.GreaterOrEqual(t, len(imports), 1)
+
+	// trim the path surrounded by quotes
+	importPath := strings.Trim(imports[0].Path.Value, "\"")
+
+	// check if present in exceptionMap
+	plugin, ok := exceptionMap[importPath]
+	if ok {
+		return plugin
+	}
+	return filepath.Base(importPath)
+}
+
+func testBuildTags(t *testing.T, buildComment string, pluginCategory string, plugin string) {
+	tags := strings.Split(buildComment, "||")
+	// tags might contain spaces and hence trim
+	tags = stringMap(tags, strings.TrimSpace)
+
+	require.Len(t, tags, 3)
+	require.Contains(t, tags, "!custom")
+	require.Contains(t, tags, pluginCategory)
+
+	actual := getPluginBuildTag(tags, pluginCategory)
+	expected := fmt.Sprintf("%v.%v", pluginCategory, plugin)
+	require.Equal(t, expected, actual, fmt.Sprintf("build tag is invalid for %v", plugin))
```

Comment:
```suggestion
	actual := getPluginBuildTag(tags, pluginCategory)
	expected := fmt.Sprintf("%s.%s", pluginCategory, plugin)
	require.Equal(t, expected, actual, "invalid build tag")
```
(plugin name is already covered in the name of the test)